olpopova/homework/homework09/fraction.py

```diff
@@ -0,0 +1,103 @@
+from math import gcd
+
+
+class Fraction:
+
+    def __init__(self, num: int, denom: int = 1):
+
+        # handle object types
+
+        if isinstance(num, float):
+            num, _denom = num.as_integer_ratio()
+            denom = _denom
+
+        elif isinstance(num, type(self)):
+            num, _denom = num.num, num.denom
+            denom = _denom * denom
```

Comment:
<img width="430" alt="image" src="https://github.com/user-attachments/assets/ed33fd47-f666-49e1-81ad-1f780368dc71">

i prefer not to split blocks which relate to same if else, to highlight visually  they relate each other
but it is ok